system/lib/wasmfs/file.cpp

```diff
@@ -83,6 +85,17 @@ std::shared_ptr<Directory> getDir(std::vector<std::string>::iterator begin,
 #endif
     curr = directory->locked().getEntry(*it);
 
+    // Special case for the rename syscall.
+    // If one detects that old_path is a forbidden ancestor in the path of
+    // new_path, the rename operation is invalid. Ex.
+    // rename("dir", "dir/somename").
```

Comment:
This comment is in high danger of being stale and misleading. Commenting on the internal details of distant code is generally not a good idea. It does increase readability of this PR's diff, but the more important thing is all the reading of the code that will happen in the future. Those people will read this file by itself.

I don't think a comment is needed here: forbiddenAncestor is explained in the header, and this just does what that says.

Reply:
That sounds good. I was originally trying to enhance readability and also thought that since `getDir` is an internal function, any re-factoring would probably change this area of the code anyways.